app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java

```diff
@@ -681,6 +683,12 @@ public void onMediaFullScreen(@NonNull WMediaSession mediaSession, boolean aFull
             }
 
             if (mAutoSelectedProjection != VIDEO_PROJECTION_NONE && autoEnter.get()) {
+                SettingsStore settingsStore = SettingsStore.getInstance(getContext());
+                if (settingsStore.isHeadLockEnabled()) {
+                    settingsStore.setHeadLockEnabled(false);
+                    disabledHeadlockForVRVideo = true;
+                }
```

Comment:
This very same block of code is repeated trice. Let's better move it to a private method and call it in all these places.